src/fwrite.c

```diff
@@ -898,21 +912,26 @@ void fwriteMain(fwriteMainArgs args)
 #ifndef NOZLIB
       }
 #endif
-      if (ret0 == -1 || ret1 || ret2 == -1) {
+      if (ret1 || ret2 == -1) {
         // # nocov start
         int errwrite = errno; // capture write errno now in case close fails with a different errno
         CLOSE(f);
         free(buffPool);
 #ifndef NOZLIB
         free(zbuffPool);
 #endif
-        if (ret0 == -1) STOP(_("Failed to write gzip header. Write returned %d"), ret0);
-        else if (ret1) STOP(_("Failed to compress gzip. compressbuff() returned %d"), ret1);
+        if (ret1) STOP(_("Failed to compress gzip. compressbuff() returned %d"), ret1);
         else STOP(_("%s: '%s'"), strerror(errwrite), args.filename);
         // # nocov end
       }
     }
   }
+#ifndef NOZLIB
```

Comment:
Interestingly #6857 is not an issue for me when deleting this block 😕

Reply:
And if `args.is_gzip` is not `true` then we never call `init_stream` so this is actually not surprising anymore.